homeassistant/components/aqara/base.py

```diff
@@ -0,0 +1,278 @@
+    def min_scaled(self) -> float:
+        """Return the min scaled."""
+        return self.scale_value(self.min)
+
+    @property
+    def step_scaled(self) -> float:
+        """Return the step scaled."""
+        return self.scale_value(self.step)
+
+    def scale_value(self, value: float | int) -> float:
+        """Scale a value."""
+        return value * 1.0 / (10**self.scale)
+
+    def scale_value_back(self, value: float | int) -> int:
+        """Return raw value for scaled."""
+        return int(value * (10**self.scale))
+
+    def remap_value_to(
+        self,
+        value: float,
+        to_min: float | int = 0,
+        to_max: float | int = 255,
+        reverse: bool = False,
+    ) -> float:
+        """Remap a value from this range to a new range."""
+        return remap_value(value, self.min, self.max, to_min, to_max, reverse)
+
+    def remap_value_from(
+        self,
+        value: float,
+        from_min: float | int = 0,
+        from_max: float | int = 255,
+        reverse: bool = False,
+    ) -> float:
+        """Remap a value from its current range to this range."""
+        return remap_value(value, from_min, from_max, self.min, self.max, reverse)
+
+    @classmethod
+    def from_json(cls, data: str) -> IntegerTypeData:
+        """Load JSON string and return a IntegerTypeData object."""
+        return cls(**json.loads(data))
+
+
+class DeviceValueRange(SimpleNamespace):
+    """device's value range.
+
+    Attributes:
+        type(str): value's type, which may be Boolean, Integer, Enum, Json
+        values(dict): value range
+    """
+
+    type: str
+    values: str
+
+
+@dataclass
+class EnumTypeData:
+    """Enum Type Data."""
+
+    range: list[str]
+
+    @classmethod
+    def from_json(cls, data: str) -> EnumTypeData:
+        """Load JSON string and return a EnumTypeData object."""
+        return cls(**json.loads(data))
+
+
+class AqaraEntity(Entity):
+    """Aqara base device."""
+
+    _attr_should_poll = False
+
+    def __init__(self, point: AqaraPoint, device_manager: AqaraDeviceManager) -> None:
+        """Init AqaraHaEntity."""
+        self._attr_unique_id = f"Aqara.{point.id}"
+        self.point = point
+        self.device_manager = device_manager
+        device = self.get_aqara_device()
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, string_dot_to_underline(self.point.did))},
+            manufacturer="Aqara",
+            name=f"{device.device_name}({self.point.did})",
+            model=device.model,
+            suggested_area=f"{device.position_name}",
+        )
+
+    def get_aqara_device(self) -> AqaraDevice:
+        """Get the device of the point."""
+        return self.device_manager.get_device(self.point.did)
+
+    @property
+    def name(self) -> str | None:
+        """Return Aqara device name."""
+        if isinstance(self.point.name, str) and self.point.name != "":
+            return f"{self.point.name}({self.point.did})"
```

Comment:
The entity should always prefer the entity description set name.
This statement causes all entities for the same device to have the exact same name. I highly doubt that that is wished for.

Reply:
No，different entity for the same device will have different names，the name is get from aqara backend server。
This is one screen shot i tested for two entitys from one same device。
![image](https://user-images.githubusercontent.com/4150132/172317498-672ac89e-b168-429b-9b9f-2340a6e614a3.png)

Reply:
ok.thanks. i updated.  device name  be prefixed to the entity description name when do device discovery

Reply:
![image](https://user-images.githubusercontent.com/4150132/174986186-2bf1fb33-a837-4ee6-9b3e-4ed351ad6dd6.png)
This is the result I tested before commit